src/js/components/modals/UpdateConfigModal.js

```diff
@@ -0,0 +1,249 @@
+import classNames from 'classnames';
+import {Modal} from 'reactjs-components';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import AdvancedConfig from '../AdvancedConfig';
+import CosmosErrorMessage from '../CosmosErrorMessage';
+import CosmosPackagesStore from '../../stores/CosmosPackagesStore';
+import InternalStorageMixin from '../../mixins/InternalStorageMixin';
+import Service from '../../structs/Service';
+import StringUtil from '../../utils/StringUtil';
+import UniversePackage from '../../structs/UniversePackage';
+import Util from '../../utils/Util';
+
+const METHODS_TO_BIND = [
+  'handleAdvancedFormChange',
+  'handleConfigSave',
+  'handleErrorConfigEditClick',
+  'handleModalClose',
+  'getConfigSubmit'
+];
+
+class UpdateConfigModal extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.internalStorage_set({
+      hasFormErrors: false,
```

Comment:
why are these in the `internalStorage`?

i've always been curious about why we've had this hatred vs state (talking about our general application, not just this component). apparently, `state` is for things that change over time. these all fit that description. so why not just use what `React` gives us?

having to `forceUpdate` not only ignores `shouldComponentUpdate`, it's also error prone. plus, you have to keep typing in `internalStorage_get` / `internalStorage_set`